test/__snapshots__/test.js.snap

```diff
@@ -523,47 +545,255 @@ const Impl = require(\\"../implementations/Factory.js\\");
 "
 `;
 
-exports[`LegacyArrayClass.webidl 1`] = `
+exports[`Global.webidl 1`] = `
 "\\"use strict\\";
 
 const conversions = require(\\"webidl-conversions\\");
 const utils = require(\\"./utils.js\\");
 
 const impl = utils.implSymbol;
 
-function LegacyArrayClass() {
-  throw new TypeError(\\"Illegal constructor\\");
+class Global {
+  constructor() {
+    throw new TypeError(\\"Illegal constructor\\");
+  }
 }
+Object.defineProperties(Global.prototype, {
+  [Symbol.toStringTag]: {
+    value: \\"Global\\",
+    configurable: true,
+    enumerable: false,
+    writable: false
+  }
+});
+Object.defineProperties(Global, {
+  prototype: {
+    configurable: false,
+    enumerable: false,
+    writable: false
+  }
+});
+const iface = {
+  // When an interface-module that implements this interface as a mixin is loaded, it will append its own \`.is()\`
+  // method into this array. It allows objects that directly implements *those* interfaces to be recognized as
+  // implementing this mixin interface.
+  _mixedIntoPredicates: [],
+  is(obj) {
+    if (obj) {
+      if (utils.hasOwn(obj, impl) && obj[impl] instanceof Impl.implementation) {
+        return true;
+      }
+      for (const isMixedInto of module.exports._mixedIntoPredicates) {
+        if (isMixedInto(obj)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  },
+  isImpl(obj) {
+    if (obj) {
+      if (obj instanceof Impl.implementation) {
+        return true;
+      }
 
-Object.setPrototypeOf(LegacyArrayClass.prototype, Array.prototype);
+      const wrapper = utils.wrapperForImpl(obj);
+      for (const isMixedInto of module.exports._mixedIntoPredicates) {
+        if (isMixedInto(wrapper)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  },
+  convert(obj, { context = \\"The provided value\\" } = {}) {
+    if (module.exports.is(obj)) {
+      return utils.implForWrapper(obj);
+    }
+    throw new TypeError(\`\${context} is not of type 'Global'.\`);
+  },
 
-Object.defineProperty(LegacyArrayClass, \\"prototype\\", {
-  value: LegacyArrayClass.prototype,
-  writable: false,
-  enumerable: false,
-  configurable: false
-});
+  create(constructorArgs, privateData) {
+    let obj = Object.create(Global.prototype);
+    obj = this.setup(obj, constructorArgs, privateData);
+    return obj;
+  },
+  createImpl(constructorArgs, privateData) {
+    let obj = Object.create(Global.prototype);
+    obj = this.setup(obj, constructorArgs, privateData);
+    return utils.implForWrapper(obj);
+  },
+  _internalSetup(obj) {
+    Object.defineProperties(
+      obj,
+      Object.getOwnPropertyDescriptors(
```

Comment:
This indirection seems unfortunate; any way to avoid it?

Reply:
I could do `defineProperties(getOwnPropertyDescriptors()); defineProperties();` rather than `defineProperties(getOwnPropertyDescriptors(defineProperties()));` as it is currently, but there aren't many ways of avoiding the `getOwnPropertyDescriptors()` because I want to use the method syntax to reduce even more `defineProperties()`.

Reply:
Maybe Object.assign + defineProperties?

Reply:
`Object.assign` will kill accessors, and accessors can't be specified through `defineProperties` due to the requirement that `Object.getOwnPropertyDescriptor(Interface.prototype, "accessor").get.name === "get accessor"` (of course, without another layer of `defineProperty("name")`).